BEIMA.Client/cypress/integration/AddDeviceTypePage.spec.js

```diff
@@ -0,0 +1,90 @@
+/// <reference types="cypress" />
+
+describe("Verify Buttons on Add Device Page", () => {
+  it('Check for Add Device Type Button', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get("#addDeviceType").contains('Add Device Type')
+  })
+
+  it('Check for Add Field Button', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get("#addField").contains('Add Field')
+  })
+})
+
+describe("Verify Data can be entered into fields", () => {
+  it('Enter data into Name field', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get('#Name').type("new type")
+  })
+  it('Enter data into Description field', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get('#Description').type("newly added type")
+  })
+  it('Enter data into Device Type Notes field', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
+    cy.get("[id='Device Type Notes']").type("meter from SEL")
+  })
+})
+
+describe("Verify Data in fields is cleared when Add Device is selected", () => {
+  it('Enter data, click Add Device, verify fields are empty', () => {
+    cy.visit('http://localhost:3000/addDeviceType')
```

Comment:
If it's not too complex, maybe we could find a way to keep the webpage loaded between various tests if they don't interfere with one another. Could possibly use some sort of setup function if cypress has that, idk. Mostly just trying to think of ways to keep the testing time down.

Reply:
Looking through the cypress docs, I didn't find an easy way for us to keep it open. We can't call `cy.visit()` outside of an `it` block